src/qutip_cupy/cupy_dense.py

```diff
@@ -0,0 +1,88 @@
+import cupy as cp
+import numbers
+from qutip.core import data
+
+
+class CuPyDense(data.Data):
```

Comment:
The `CuPyDense` class docstring should detail the arguments to `__init__` and how `CuPyDense` should be used and have a very short explanation to how the data class works under the hood (e.g.  it stores data in arrays on the GPU using CuPy, it copies data in these circumstances, etc).